drivers/hmis/app/graphql/mutations/create_assessment.rb

```diff
@@ -0,0 +1,64 @@
+module Mutations
+  class CreateAssessment < BaseMutation
+    argument :enrollment_id, ID, required: false
+    argument :assessment_role, String, required: true
+
+    field :assessment, Types::HmisSchema::Assessment, null: true
+    field :errors, [Types::HmisSchema::ValidationError], null: false
+
+    def validate_input(enrollment: nil, definition: nil)
+      errors = []
+      errors << InputValidationError.new('Enrollment must exist', attribute: 'enrollment_id') unless enrollment.present?
+      errors << InputValidationError.new('Cannot get definition for assessment role', attribute: 'assessment_role') if enrollment.present? && definition.nil?
+      errors
+    end
+
+    def resolve(enrollment_id:, assessment_role:)
+      user = hmis_user
+
+      enrollment = Hmis::Hud::Enrollment.find_by(id: enrollment_id)
+      form_definition = enrollment&.project&.present? ? Hmis::Form::Definition.find_definition_for_project(enrollment.project, role: assessment_role) : nil
+
+      errors = validate_input(enrollment: enrollment, definition: form_definition)
+
+      return { assessment: nil, errors: errors } if errors.present?
+
+      assessment_attrs = {
+        data_source_id: user.data_source_id,
+        user_id: user.user_id,
+        personal_id: enrollment.personal_id,
+        assessment_id: Hmis::Hud::Assessment.generate_assessment_id,
+        # START These values will probably end up being nullable, but aren't now
+        assessment_date: Date.today,
```

Comment:
I believe we should require `assessment_date` and `assessment_location` always.  `assessment_location` can default to the project name.  The other 3 will probably need something like `99` or `999` added that we can call "For Local Use" in the enum.

Reply:
This is fixed as we discussed in the call